src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/NonAssociationsMergeService.kt

```diff
@@ -54,26 +49,43 @@ class NonAssociationsMergeService(
           nonAssociationsRepository.findByFirstPrisonerNumberAndSecondPrisonerNumber(
             firstPrisonerNumber = nonAssociation.firstPrisonerNumber,
             secondPrisonerNumber = newPrisonerNumber,
-          ).let {
-            if (it != null) {
-              nonAssociationsRepository.delete(nonAssociation)
-              log.info("Deleted Non association record ${nonAssociation.id}")
-              it.comment = nonAssociation.comment // replace comment??
-              updatedRecords.add(it)
-            } else {
-              if (newPrisonerNumber == nonAssociation.firstPrisonerNumber) {
-                nonAssociationsRepository.delete(nonAssociation)
-                log.info("Deleted Non association record ${nonAssociation.id} as same prisoner both ways")
-              } else {
-                log.info("Updated Non association record ${nonAssociation.id} from ${nonAssociation.secondPrisonerNumber} to $newPrisonerNumber")
-                nonAssociation.secondPrisonerNumber = newPrisonerNumber
-                updatedRecords.add(nonAssociation)
-              }
+          ).let { duplicateRecord ->
+            merge(
+              nonAssociation,
+              duplicateRecord,
+              newPrisonerNumber,
+              nonAssociation.firstPrisonerNumber,
+              false,
+            )?.let { updatedRecord ->
+              updatedRecords.add(updatedRecord)
             }
           }
         }
       }
 
     return updatedRecords.toList()
   }
+
+  private fun merge(
+    nonAssociation: NonAssociation,
+    duplicateRecord: NonAssociation?,
+    newPrisonerNumber: String,
+    otherPrisonerNumber: String,
+    primary: Boolean,
+  ) =
+    if (duplicateRecord != null) {
+      log.info("Deleting non-association record ${nonAssociation.id} - Duplicate")
+      nonAssociationsRepository.delete(duplicateRecord)
```

Comment:
Did you mean `nonAssociationsRepository.delete(nonAssociation)` here? Sorry if this is correct.